lib/stats/stats-query.c

```diff
@@ -167,12 +168,7 @@ _format_selected_counters(GList *counters, StatsFormatCb format_cb, gpointer res
 static void
 _reset_selected_counters(GList *counters)
 {
-  GList *c;
-  for (c = counters; c; c = c->next)
-    {
-      StatsCounterItem *counter = c->data;
-      stats_counter_set(counter, 0);
-    }
+  stats_control_reset_counters(counters, FALSE);
```

Comment:
this is implementing an O(N*M) algorithm where N is the number of all counters, and M is the number of queried counters. That can be _very_ slow in cases where we have 10k stat counters and 100 queried ones. I didn't measure this.

We did have performance issues that stemmed from stats queries so I would be a cautious here.

Reply:
what other lookup method you could suggest here that would perform better on a list
this would be a bottleneck only, if the query has a big result set that I guess usually is not the case